sync/syncer.go

```diff
@@ -201,29 +238,71 @@ func (s *Syncer) Synchronise() {
 	currentSyncLayer := s.lValidator.ValidatedLayer() + 1
 	if s.lastTickedLayer() == 1 { // skip validation for first layer
 		s.Info("Not syncing in layer 1")
+		s.setP2pSynced(true)
 		return
 	}
-	if s.IsSynced() {
-		lyr, err := s.GetLayer(types.LayerID(currentSyncLayer))
+	if s.WeaklySynced() {
+		s.Info("Node is synced. Going to validate layer %v", currentSyncLayer)
+
+		lyr, err := s.lProvider.GetLayer(currentSyncLayer)
 		if err != nil {
-			s.With().Error("failed getting layer even though IsSynced is true", log.Err(err))
+			s.With().Error("failed getting layer even though we are weakly-synced", log.Err(err))
```

Comment:
shouldn't we panic if this happens?

Reply:
Does it break our assumptions? Yes
Did something go awfully wrong? Yes
Can we continue? not entirely in an honest way

I think it is a panic.
@barakshani ?

Reply:
Should be panic, however there's an edge case with the very first layer.

Reply:
Fixed